pkg/build/strategies/sti/sti.go

```diff
@@ -401,7 +404,7 @@ func (b *STI) Execute(command string, config *api.Config) error {
 		Image:           config.BuilderImage,
 		Stdout:          outWriter,
 		Stderr:          errWriter,
-		PullImage:       config.ForcePull,
+		PullImage:       config.BuilderPullPolicy == api.PullAlways || config.BuilderPullPolicy == api.PullIfNotPresent,
```

Comment:
doesn't this always result in a Pull for PullIfNotPresent?  In the case of PullIfNotPresent, don't you need to manually check if the image exists locally first, and then set this to true or false accordingly?